### DISCUSSION TITLE
Could we have DrawBoxShadows api, or DrawGeometry with BoxShadows parameter?

### DISCUSSION THREAD

**Discussion by dbriard** [AvaloniaUI/Avalonia#13279] [Q&A]
It would be very handy to have DrawGeometry with BoxShadows parameter in the same way as DrawRectangle? Or something like DrawBoxShadows (with clipping lile DrawRectangle).

The idea is to be able to draw such geometry without relying on ICutstomDrawOperation like I did:
![image](https://github.com/AvaloniaUI/Avalonia/assets/3595443/140b7dff-6e6b-4e93-831c-f99c76cdf3c9)

Thanks a lot!

**Comment by maxkatz6** [comment]
In Avalonia (and CSS for that matter) box shadows do make sense only with rectangles.
And we already have an API for that:
https://github.com/AvaloniaUI/Avalonia/blob/43f1b9d8ca9389fea51a5463b003af325b171349/src/Avalonia.Base/Media/ImmediateDrawingContext.cs#L119

This DrawRectangle accepts box shadows argument, and can be used to achieve the same result.

**Comment by dbriard** [comment]
> In Avalonia (and CSS for that matter) box shadows do make sense only with rectangles. And we already have an API for that:
> 
> https://github.com/AvaloniaUI/Avalonia/blob/43f1b9d8ca9389fea51a5463b003af325b171349/src/Avalonia.Base/Media/ImmediateDrawingContext.cs#L119
> 
> This DrawRectangle accepts box shadows argument, and can be used to achieve the same result.

Hi @maxkatz6 
I agree that in 90% of cases, Border and DrawRectangle can do the job, but as recent UI design languages such as Fluent or Fluent2 start to add a **tail** to their popup/tooltip/teaching tip/etc. (and it look great), that would be nice to be able to reproduce the effect in Avalonia more without relying to custom draw operation.  
By looking at the code of DrawRectangle, shadows can be applied to any geometry without trouble, so why not... for me at least that would be useful as I think border with tail is a valuable design element.
Moreover DropShadow to any geometry is available in design tool such as Adobe XD and Figma and designers are using for more than just rectangles.

A few examples:
![image](https://github.com/AvaloniaUI/Avalonia/assets/3595443/95e60892-913f-4803-ba4d-c3e461442abf)
![image](https://github.com/AvaloniaUI/Avalonia/assets/3595443/d5e61e9e-940e-4368-9f74-21e29096e701)

Another real case example to illustrate this request: this handle look weird without shadow, ideally the shadow must be applied to it, but it is not a rectangle or rounded rectangle.
![image](https://github.com/AvaloniaUI/Avalonia/assets/3595443/2a6b63c9-1954-48a5-958a-0e24ef9083ca)
